enhancements/windows-containers/health-management.md

```diff
@@ -0,0 +1,258 @@
+  }
+]
+```
+
+### Node configuration: WMCO responsibilities
+
+1) For a given Machine or BYOH VM, WMCO copies the payload and WMCD binary to the VM.
+2) WMCO runs WMCD with the bootstrap command, this results in the node object being created.
+3) WMCO annotates the Node object with the `desiredVersion` annotation.
+4) The value of this is set to the current WMCO version.
+5) WMCO starts WMCD as a service.
+6) Reconciliation can successfully return at this point.
+
+### Node configuration: WMCD responsibilities
+
+For node configuration WMCD will have two separate responsibilities:
+
+#### Bootstrap command
+
+When run with the bootstrap command WMCD will do the steps required to ensure that Node is created
+for the VM. This functionality is currently implemented within WMCB with the `init-kubelet` command.
+
+#### Running as a Windows service
+
+WMCD will have an informer for both ConfigMap and Node events. When it is detected that either the desired version
+has been set for the Node, or the ConfigMap specified by the desired version annotation has been created, WMCD will
+configure the Windows services to the specifications listed in relevant the ConfigMap. If an error occurs, an event
+will be generated against the Node object.
+
+When configuring a service, WMCD will ensure that the service exists as specified in the ConfigMap. In order to
+properly do this, it must evaluate the variables listed within the ConfigMap. This will be done using the Node object,
+which can be achieved by searching for a Node with an internal IP equivalent to the IP of the VM, and the powershell
+scripts copied to the VM by WMCO.
+
+On a separate thread, WMCD will be periodically polling the state of Windows services. If there is any change in the
+expected state, the services will be created or modified in order to restore expected state. As multiple threads can
+be modifying Windows service state, the appropriate thread safety procedures must be followed.
+
+### Node de-configuration: WMCO responsibilies
+
+When a user indicates to WMCO that it wishes to remove a BYOH node from the cluster, the following procedure will be
+kicked off.
+
+1) WMCO stops the WMCD service on the VM. The dependent services will be automatically stopped as well.
+2) WMCO runs the deconfigure command of WMCD.
+3) WMCO removes the files it copied to the VM.
+
+### Node de-configuration: WMCD responsibilities
+
+When the cleanup command is ran, WMCD will remove the services listed in the ConfigMap used to configure the VM.
+It will then delete the VM's Node object.
+
+### User Stories
+
+User stories can be found within the [health management epic](https://issues.redhat.com/browse/WINC-657).
+
+### API Extensions
+
+N/A
+
+### Risks and Mitigations
+
+* If WMCD has an unrecoverable error that results in the process continuously crashing, the services that it manages
+  will go down, due to them being set as dependents. With the kubelet down, the Node will be unreachable and the Ready
```

Comment:
Do you mean the `NotReady` or whatever state that is?

Reply:
WMCD is a support service and not related to essential Kubernetes services. If it crashes but Kubernetes services are still up and running, we should let them run.  If you make them a dependent service, they will be stopped if there is an issue with WMCD.

Reply:
Interesting. @selansen are you saying that WMCD should not control `kubelet` service state in general or just while facing an error ?

Reply:
>> Interesting. @selansen are you saying that WMCD should not control kubelet service state in general or just while facing an error ?
Just saying when WMCD crashes and other services like kubelet are running without any issue, why should we stop them. Then node will go to `not ready` state.